### PR TITLE
[115] Add DDTP environment variables to terraform

### PR DESCRIPTION
**Pull request by felixtheflex**
### Context

Add new DDTP environment variables through terraform

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -85,6 +85,13 @@ jobs:
     needs: build
     env:
       SHA_TAG: ${{needs.build.outputs.sha_tag}}
+      TF_VAR_paas_settings__basic_auth__password: ${{ secrets.TF_VAR_PAAS_SETTINGS__BASIC_AUTH__PASSWORD }}
```
Comment (saliceti) [inline .github/workflows/build.yml:88]: Can we move this inside the step which requires the secrets?

**Comment by saliceti** [comment]
Please remove the comments and run terraform fmt

**Comment by defong** [comment]
[Add DTTP environment variables](https://trello.com/c/GqiYlYgO/115-add-dttp-environment-variables)

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -114,6 +121,10 @@ jobs:
         TF_VAR_paas_password: ${{ secrets.TF_VAR_PAAS_PASSWORD }}
         TF_VAR_paas_user: ${{ secrets.TF_VAR_PAAS_USER }}
         TF_VAR_paas_log_url: ${{ secrets.TF_VAR_PAAS_LOGGING_ENDPOINT_PORT }}
+        #TF_VAR_paas_settings__basic_auth__password: ${{ secrets.TF_VAR_PAAS_SETTINGS__BASIC_AUTH__PASSWORD }} #secrets
+        #TF_VAR_paas_settings__dttp__client_secret:  ${{ secrets.TF_VAR_PAAS_SETTINGS__DTTP__CLIENT_SECRET }}  # secrets
```
Comment (defong) [inline .github/workflows/build.yml:125]: Do we need these?

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -123,3 +134,5 @@ jobs:
...
```
Comment (defong) [inline .github/workflows/build.yml:138]: Do we need these?

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -85,6 +85,13 @@ jobs:
     needs: build
     env:
       SHA_TAG: ${{needs.build.outputs.sha_tag}}
+      TF_VAR_paas_settings__basic_auth__password: ${{ secrets.TF_VAR_PAAS_SETTINGS__BASIC_AUTH__PASSWORD }}
+      TF_VAR_paas_settings__dttp__client_secret:  ${{ secrets.TF_VAR_PAAS_SETTINGS__DTTP__CLIENT_SECRET }}
```
Comment (defong) [inline .github/workflows/build.yml:89]: Can we organise this one to sit along the rest of `TF_VAR_paas_settings__dttp*`?

File: terraform/workspace-variables/qa.tfvars [hunk 5]
```diff
@@ -11,4 +11,11 @@ paas_web_app_memory              = 512
 paas_worker_app_instances        = 1
 paas_worker_app_memory           = 512
 
+#paas_settings__basic_auth__username = "register"
+#paas_settings__basic_auth__password = ""  #secrets
+#paas_settings__dttp__client_id = ""
+#paas_settings__dttp__tenant_id = ""
+#paas_settings__dttp__client_secret = "" # secrets
+#paas_settings__dttp__api_base_url = ""
```
Comment (defong) [inline terraform/workspace-variables/qa.tfvars:19]: Do we need these?

Comment (felixtheflex) [reply terraform/workspace-variables/qa.tfvars]: they've been removed already.

File: .github/workflows/build.yml (hunk 4 above)
Comment (felixtheflex) [reply .github/workflows/build.yml]: done already..

File: .github/workflows/build.yml [hunk 6]
```diff
@@ -105,21 +105,20 @@ jobs:
         TF_VAR_paas_app_docker_image: ${{env.DOCKER_REPO}}:${{env.SHA_TAG}}
         ARM_ACCESS_KEY: ${{ secrets.ARM_ACCESS_KEY }}
 
-
-    - name: Terraform plan
-      run: terraform plan -var-file=terraform/workspace-variables/qa.tfvars terraform
+    - name: Terraform Plan and Apply
+      run: |
+        terraform plan -var-file=terraform/workspace-variables/qa.tfvars terraform
+        terraform apply -var-file=terraform/workspace-variables/qa.tfvars -auto-approve terraform
```
Comment (defong) [inline .github/workflows/build.yml:111]:
```suggestion
        terraform apply -var-file=terraform/workspace-variables/qa.tfvars -auto-approve terraform
```

There seems to be very little point with plan

Comment (saliceti) [reply .github/workflows/build.yml:111]: plan shows you the changes that will be applied. it's good to keep running it for troubleshooting

**Review by saliceti — APPROVED** [review]
Thank you

**Review by vigneshmsft — APPROVED** [review]

**Pull request merged**